apps/office/src/lib/server/api.js

```diff
@@ -0,0 +1,126 @@
+import { BAG_API_KEY, API_BASE_URL } from '$env/static/private';
+import { error } from '@sveltejs/kit';
+import { normalizeURL } from '$utils';
+
+// Function to normalize API base URL
+const apiUrl = (path) => normalizeURL(API_BASE_URL) + path;
+
+export const bagApi = {
+    getBuildingByPostcodeAndNumber: async (postcode, huisnummer, page = 1, pageSize = 20) => {
+        if (!BAG_API_KEY) {
+            throw new Error('BAG API Key not configured');
+        }
+        const bagUrl = `https://api.bag.kadaster.nl/lvbag/individuelebevragingen/v2/adressenuitgebreid?postcode=${postcode}&huisnummer=${huisnummer}&page=${page}&pageSize=${pageSize}`;
+        try {
+            const res = await fetch(bagUrl, {
+                method: 'GET',
+                headers: {
+                    'accept': 'application/hal+json',
+                    'Accept-Crs': 'epsg:28992',
+                    'X-Api-Key': BAG_API_KEY
+                }
+            });
+            if (res.ok) {
+                const data1 = await res.json();
+                return data1._embedded.adressen[0];
+            } else {
+                throw error(res.status, 'Failed to fetch from BAG API');
```

Comment:
De ingebouwde error functie throwed zelf een Error. Geldt voor alle error functies in dit bestand.

```suggestion
                error(res.status, 'Failed to fetch from BAG API');
```